framework/platform/HLAPI/CMakeLists.txt

```
@@ -0,0 +1,43 @@
+project(hlapi VERSION ${prplmesh_VERSION})
+
+message("${BoldWhite}Preparing ${BoldGreen}${PROJECT_NAME}${BoldWhite} for the ${BoldGreen}${TARGET_PLATFORM}${BoldWhite} platform${ColourReset}")
+
+# Set the base path for the current module
+set(MODULE_PATH ${CMAKE_CURRENT_LIST_DIR})
+
+# Common Sources
+file(GLOB_RECURSE hlapi_sources ${MODULE_PATH}/*.c*)
+
+# Include AMBIORIX in the build
+message(STATUS "ENABLE_AMBIORIX: ${ENABLE_AMBIORIX}")
+find_package(amxb REQUIRED)
+find_package(amxc REQUIRED)
+find_package(amxd REQUIRED)
+find_package(amxj REQUIRED)
+find_package(amxo REQUIRED)
+find_package(amxp REQUIRED)
+list(APPEND HLAPI_LIBS amxb amxc amxd amxj amxo amxp)
+
+
+# Build the library
+add_library(${PROJECT_NAME} ${hlapi_sources})
+set_target_properties(${PROJECT_NAME} PROPERTIES VERSION ${prplmesh_VERSION} SOVERSION ${prplmesh_VERSION_MAJOR})
+set_target_properties(${PROJECT_NAME} PROPERTIES LINK_FLAGS "-Wl,-z,defs")
+target_link_libraries(${PROJECT_NAME} PRIVATE ${HLAPI_LIBS})
+
+# Install
+target_include_directories(${PROJECT_NAME}
+    PUBLIC
+        $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/include>
+        $<BUILD_INTERFACE:${CMAKE_CURRENT_BINARY_DIR}/include>
```

AFAIK: the binary directory should go to /lib
```suggestion
        $<BUILD_INTERFACE:${CMAKE_CURRENT_BINARY_DIR}/lib>
```

---

Accepted. Resolving.